lib/sdr_client/deposit/upload_files.rb

```diff
@@ -8,40 +8,43 @@ module Deposit
     class UploadFiles
       BLOB_PATH = '/v1/direct_uploads'
 
-      # @param [Hash<String,Files::DirectUploadRequest>] the metadata for uploading the files
+      # @param [Hash<String,Files::DirectUploadRequest>] file_metadata map of relative filepaths to file metadata
+      # @param [Hash<String,String>] filepath_map map of relative filepaths to absolute filepaths
       # @param [Logger] logger the logger to use
       # @param [Connection] connection
-      def self.upload(file_metadata:, logger:, connection:)
-        new(file_metadata: file_metadata, logger: logger, connection: connection).run
+      def self.upload(file_metadata:, filepath_map:, logger:, connection:)
+        new(file_metadata: file_metadata, filepath_map: filepath_map, logger: logger, connection: connection).run
       end
 
-      # @param [Hash<String,Files::DirectUploadRequest>] the metadata for uploading the files
+      # @param [Hash<String,Files::DirectUploadRequest>] file_metadata map of relative filepaths to file metadata
+      # @param [Hash<String,String>] filepath_map map of relative filepaths to absolute filepaths
       # @param [Logger] logger the logger to use
       # @param [Connection] connection
-      def initialize(file_metadata:, logger:, connection:)
+      def initialize(file_metadata:, filepath_map:, logger:, connection:)
         @file_metadata = file_metadata
+        @filepath_map = filepath_map
         @logger = logger
         @connection = connection
       end
 
       # @return [Array<Files::DirectUploadResponse>] the responses from the server for the uploads
       def run
-        file_metadata.map do |filename, metadata|
+        file_metadata.map do |filepath, metadata|
           direct_upload(metadata.to_json).tap do |response|
-            # ActiveStorage modifies the filename provided in response, so setting here.
-            response.filename = filename
-            upload_file(filename: filename,
+            # ActiveStorage modifies the filename provided in response, so setting here with the relative filename
+            response.filename = filepath
```

Comment:
This really is the crucial change -- getting the correct filepath in the upload response so that all of the files can be stitched back together.

Reply:
Good eye! Time well spent. 😄